src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/InstalledAndTransitivePackageFeed.cs

```diff
@@ -29,16 +29,84 @@ public override async Task<SearchResult<IPackageSearchMetadata>> ContinueSearchA
+            var transitiveItems = await GetMetadataForPackagesAsync(PerformLookup(latestTransitivePackages, searchToken), searchToken.SearchFilter.IncludePrerelease, cancellationToken);
 
-            IPackageSearchMetadata[] installedItems = await GetMetadataForPackagesAndSortAsync(PerformLookup(_installedPackages.GetLatest(), searchToken), searchToken.SearchFilter.IncludePrerelease, cancellationToken);
-            IPackageSearchMetadata[] transitiveItems = await GetMetadataForPackagesAndSortAsync(PerformLookup(transitivePkgsWithOrigins.GetLatest(), searchToken), searchToken.SearchFilter.IncludePrerelease, cancellationToken);
-            IPackageSearchMetadata[] items = installedItems.Concat(transitiveItems).ToArray();
+            // Get metadata from identity for the rest of the installed and transitive packages
+            var remainingInstalledItems = await GetRemainingPackagesMetadataAsync(_installedPackages, searchToken, cancellationToken);
+            var remainingTransitiveItems = await GetRemainingPackagesMetadataAsync(transitivePkgsWithOrigins, searchToken, cancellationToken);
+
+            // Combine the installed and transitive packages
+            IPackageSearchMetadata[] items = installedItems
+                .Concat(remainingInstalledItems)
+                .Concat(transitiveItems)
+                .Concat(remainingTransitiveItems)
+                .ToArray();
 
             return CreateResult(items);
         }
 
+        private static IEnumerable<PackageCollectionItem> GetTransitivePackagesWithOrigins(IEnumerable<PackageCollectionItem> transitivePackages)
+        {
+            return transitivePackages
+                .Where(t => t.PackageReferences.All(x => x is ITransitivePackageReferenceContextInfo y && y.TransitiveOrigins.Any()));
+        }
+
+        private static IEnumerable<PackageCollectionItem> GetLatestPackages(IEnumerable<PackageCollectionItem> packages)
+        {
+            return packages
+                .GroupById()
+                .Select(g => g.OrderByDescending(x => x.Version).First());
+        }
+
+        private static async Task<IPackageSearchMetadata[]> GetRemainingPackagesMetadataAsync(IEnumerable<PackageCollectionItem> packages, FeedSearchContinuationToken searchToken, CancellationToken cancellationToken)
+        {
+            var remainingPackages = packages
+                .GroupById()
+                .Select(g => g.OrderByDescending(x => x.Version).Skip(1)) // Skip 1 to get the remaining packages, latest package is already processed
+                .SelectMany(s => s);
+
+            return await GetMetadataFromIdentityForPackagesAsync(PerformLookup(remainingPackages, searchToken), searchToken.SearchFilter.IncludePrerelease, cancellationToken);
+        }
+
+        internal static async Task<IPackageSearchMetadata[]> GetMetadataFromIdentityForPackagesAsync<T>(T[] packages, bool includePrerelease, CancellationToken cancellationToken) where T : PackageIdentity
```

Comment:
Unless there's any external references to this

```suggestion
        private static async Task<IPackageSearchMetadata[]> GetMetadataFromIdentityForPackagesAsync<T>(T[] packages, bool includePrerelease, CancellationToken cancellationToken) where T : PackageIdentity
```